### PR TITLE
tartufo: update to use pygit2 1.15.1

### PR DESCRIPTION
**Pull request by chenrui333**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

followup #166767

### CODE REVIEW

**Comment by chenrui333** [comment]
```
    File "/home/linuxbrew/.linuxbrew/Cellar/tartufo/5.0.0_3/libexec/lib/python3.12/site-packages/tartufo/util.py", line 305, in extract_commit_metadata
      "commit_hash": commit.hex,
                     ^^^^^^^^^^
  AttributeError: '_pygit2.Commit' object has no attribute 'hex'
```

**Comment by carlocab** [comment]
We should just use the `pygit2` formula here. We have that so that formulae that use `pygit2` don't need to be recompiled whenever `libgit2` is updated.

File: Formula/t/tartufo.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ class Tartufo < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:   "9f44c5168bfefe9795d915ea7b527d1b67f39500e1186cbb6a4bcaa0b329ee7a"
   end
 
-  depends_on "libgit2@1.7"
+  depends_on "libgit2"
```
Comment (carlocab) [inline Formula/t/tartufo.rb:22]:
```suggestion
  depends_on "pygit2"
```

**Review by carlocab — APPROVED** [review]

**Review by branchv — APPROVED** [review]

File: Formula/t/tartufo.rb (hunk 1 above)
Comment (branchvincent) [reply Formula/t/tartufo.rb]: lgtm

**Pull request merged**